dra-evolution/pkg/api/capacity_types.go

```diff
@@ -69,28 +74,94 @@ type ResourcePoolSpec struct {
 	// them) empty pool.
 }
 
+const ResourcePoolMaxSharedCapacity = 128
 const ResourcePoolMaxDevices = 128
 
+type DeviceShape struct {
+	// Attributes defines the attributes of this device shape.
+	// The name of each attribute must be unique.
+	//
+	// Must not have more than 32 entries.
+	//
+	// +listType=atomic
+	// +optional
+	Attributes []DeviceAttribute `json:"attributes,omitempty" protobuf:"bytes,3,opt,name=attributes"`
+
+	// Partitions defines the set of partitions into which this device shape
+	// may be allocated. If not populated, then the device shape is always
+	// consumed in its entirety.
+	//
+	// +listType=atomic
+	// +optional
+	Partitions []DevicePartition `json:"partitions,omitempty"`
```

Comment:
In the "simple" case, when I just have a bunch of devices that are not partitionable, what does my resourceSlice look like? Do I need to still fill this in with just 1 device (which is actually not partitionable)?

Reply:
Edit: Nevermind, I see now -- this is not a replacement for `Device`, it is something that defines how *every* device in this slice "looks", modulo its name and any custom attributes.

Reply:
Exactly